test/test_priority_queue.jl

```diff
@@ -210,6 +210,19 @@ import Base.Order.Reverse
             @test Set(collect(pq)) == Set(["a"=>2, "c"=>1])
         end
     end
+    
+    @testset "Iteration" begin
+        io = IOBuffer()
+        pq = PriorityQueue('a'=>'A', 'b'=>'B', 'c'=>'C')
+        for (k, v) in pq
+            println(io, k, " => ", v)
+        end
+        if Sys.WORD_SIZE == 32
+            @test String(take!(copy(io))) == "b => B\nc => C\na => A\n"
+        else
+            @test String(take!(copy(io))) == "a => A\nc => C\nb => B\n"
+        end
+    end
```

Comment:
It's not really a good idea to depend on the sort order of an unordered collection.  The actual order is an implementation detail that might change.  (For example, there was talk previously of changing `Dict` to maintain insertion order of keys when iterating.  It didn't happen, but it could.)

Instead, a simpler test would probably suffice in this case.  How about just doing 

```julia
pq = PriorityQueue('a'=>'A')
@test string(dequeue_pair!(pq)) == "'a' => 'A'"
```

From your own examples in #434, this would have failed previously.